client/src/components/App.jsx

```diff
@@ -58,29 +73,28 @@ class App extends React.Component {
   }
 
   componentDidMount() {
-    this.getRoomData(this.props.selectedRoom);
+    const id = window.location.pathname.split('/')[2];
+    this.getRoomData(id);
   }
 
   render () {
```

Comment:
This may be a matter of preference but you could move your conditional render if/else statement out of your render function. This may allow your render to look a little cleaner and not have any work happening within it.

You would just create a function outside of the render and then call it within. We can talk about this if that was not clear.

Reply:
Thanks for your suggestion. I have just refactored it too. Let me know if it is better. I am not sure it made a big difference. if it is easier to read, I am glad.